src/include/sof/lib/agent.h

```diff
@@ -25,15 +28,37 @@ struct sa {
 	struct perf_cnt_data pcd;
 #endif
 	struct task work;
+	atomic_t panic_cnt;	/**< ref counter for panic_on_delay property */
+	bool panic_on_delay;	/**< emits panic on delay if true */
 };
 
 #if CONFIG_HAVE_AGENT
 
+/**
+ * Enables or disables panic on agent delay.
+ * @param enabled True for panic enabling, false otherwise.
+ */
+static inline void sa_set_panic_on_delay(bool enabled)
```

Comment:
I feel like the implementation is weird. By default panic is enabled and the reference count is 0. But when there is a DMA pipeline, should we have a _negative_ reference count? Sorry for commenting after the merge, holidays...